hypothesis-python/tests/cover/test_provisional_strategies.py

```diff
@@ -53,3 +56,21 @@ def test_is_IP6_addr(address):
     assert all(len(part) == 4 for part in as_hex)
     raw = unhexlify(address.replace(u":", u"").encode("ascii"))
     assert len(raw) == 16
+
+
+@pytest.mark.parametrize("max_length", [-1, 0, 4.0, 256])
+@pytest.mark.parametrize("max_element_length", [-1, 0, 4.0, 64, 128])
+def test_invalid_domain_arguments(max_length, max_element_length):
+    with pytest.raises(InvalidArgument):
+        domains(max_length=max_length, max_element_length=max_element_length).example()
```

Comment:
You'll need a separate test for each argument for this pattern to be exhaustive.  As-is, the test would pass even if one was not validated at all!

This also obsoletes `test_more_invalid_domain_arguments` below, so you could delete that test.

Reply:
The lower limit of `max_length` is 4, so worth a test case of `max_length=3`?

Reply:
@Zac-HD  Whoops, I forgot to delete `test_more_invalid_domain_arguments` It was part of my many, many attempts to satisfy the coverage checker till I gave up and wrote the dedicated input checker function as a static method on the class. And I’m pretty sure he parametric testing will fail cause I saw it when I accidentally left non error throwing variations in one or the other parameters. But I’ve revised this so many times I could be mistaken haha.

@alexwlchan I was using `0` as the test below the lower threshold, but if I’m still using the parametric testing, it’s easy to add, so I’ll just add it in.

Reply:
These should be fixed in the latest versions of the branch.